src/test/java/org/folio/rest/impl/NoteTypesImplTest.java

```diff
@@ -58,6 +62,169 @@ public void shouldReturn200WithNoteTypeWhenValidId() throws IOException, URISynt
+
+      for (NoteType noteType : noteTypes) {
+        NoteTypesTestUtil.insertNoteType(vertx, noteType.getId(), mapper.writeValueAsString(noteType));
+      }
+      Response response = RestAssured.given()
+        .spec(getRequestSpecification())
+        .when()
+        .get(NOTE_TYPES_ENDPOINT)
+        .then()
+        .statusCode(200)
+        .extract().response();
+
+      int totalRecords = response.path(TOTAL_RECORDS);
+      List<NoteType> noteTypeList = response.path(NOTE_TYPES);
+
+      assertEquals(3, noteTypeList.size());
+      assertEquals(3, totalRecords);
+    } finally {
+      NoteTypesTestUtil.deleteAllNoteTypes(vertx);
+    }
+  }
+
+  @Test
+  public void shouldReturn200WithNoteTypeCollectionAndIncompleteWay() throws IOException, URISyntaxException {
+    try {
+      final String stubNoteType = readFile(POST_NOTE_TYPE_JSON);
+
+      NoteTypesTestUtil.insertNoteType(vertx, ID, stubNoteType);
+
+      Response response = RestAssured.given()
+        .spec(getRequestSpecification())
+        .when()
+        .get(NOTE_TYPES_ENDPOINT + "?quer")
+        .then()
+        .statusCode(200)
+        .extract().response();
+
+      int totalRecords = response.path(TOTAL_RECORDS);
+      List<NoteType> noteTypes = response.path(NOTE_TYPES);
+
+      assertEquals(1, noteTypes.size());
+      assertEquals(1, totalRecords);
+    } finally {
+      NoteTypesTestUtil.deleteAllNoteTypes(vertx);
+    }
+  }
+
+  @Test
+  public void shouldReturn200WithEmptyNoteTypeCollection() {
+    try {
+      Response response = RestAssured.given()
+        .spec(getRequestSpecification())
+        .when()
+        .get(NOTE_TYPES_ENDPOINT)
+        .then()
+        .statusCode(200)
+        .extract().response();
+
+      int totalRecords = response.path(TOTAL_RECORDS);
+      List<NoteType> noteTypes = response.path(NOTE_TYPES);
+
+      assertEquals(0, noteTypes.size());
+      assertEquals(0, totalRecords);
+    } finally {
+      NoteTypesTestUtil.deleteAllNoteTypes(vertx);
+    }
+  }
+
+  @Test
+  public void shouldReturn400InvalidRequest() throws IOException, URISyntaxException {
+    try {
+      final String stubNoteType = readFile(POST_NOTE_TYPE_JSON);
+
+      NoteTypesTestUtil.insertNoteType(vertx, ID, stubNoteType);
+
+      RestAssured.given()
+        .spec(getRequestSpecification())
+        .when()
+        .get(NOTE_TYPES_ENDPOINT + "?query=")
+        .then()
+        .statusCode(400)
+        .extract().asString();
+    } finally {
+      NoteTypesTestUtil.deleteAllNoteTypes(vertx);
+    }
+  }
+
+  @Test
+  public void shouldReturn400InvalidLimit() throws IOException, URISyntaxException {
+    try {
+      final String stubNoteType = readFile(POST_NOTE_TYPE_JSON);
+
+      NoteTypesTestUtil.insertNoteType(vertx, ID, stubNoteType);
+
+      RestAssured.given()
+        .spec(getRequestSpecification())
+        .when()
+        .get(NOTE_TYPES_ENDPOINT + "?limit=")
```

Comment:
Can we add tests for limit set and offset set?

Reply:
Yes sure.